### DISCUSSION TITLE
Keeping newline(\n) after setValue

### DISCUSSION THREAD

**Discussion by FiveBoroughs** [react-hook-form/react-hook-form#1883] [Q&A] [answered]
How can I keep the newline (`\n`) character when using setValue?

Using the following code 
![image](https://user-images.githubusercontent.com/40718516/84793059-2bc5a980-afe4-11ea-963b-d46858d30370.png)
I get this result
![image](https://user-images.githubusercontent.com/40718516/84793111-413ad380-afe4-11ea-9b48-173d08cbfffe.png)

The newline character is present in the `text` value and `console.log` is interpreting it correctly, showing 3 lines.
Converting to Json shows `\n` is present.

But after using [setValue](https://react-hook-form.com/api#setValue), it is lost.

**Comment by bluebill1049** [comment, marked as answer]
you want to put a generic question on stack overflow? i am not quite sure about the solution, try to set an input value with DOM API, if that's not working ask a question at StackOverflow see if there is any solution.

**Comment by AHorak** [comment]
Do you have a solution?
I have exactly the same problem.

**Comment by camsloanftc** [comment]
I just wanted to come provide the solution I ended up using, which was to change from using an uncontrolled text input field, to using a controlled field.

```js
import { Controller } from "react-hook-form"
// ...

// old
<input
  style={{ display: "none" }}
  ref={register()}
  name={`steps[${index}].bodyText`}
  defaultValue={item.bodyText}
/>


//new
<Controller
  render={({ onChange, value, ref }) => { return null }}
  control={control}
  name={`steps[${index}].bodyText`}
  defaultValue={item.bodyText}
/>
                          
```

Then I can call `setValue(`steps[${index}].bodyText`, value)` from my other component (which uses markdown) and it keeps the newlines in the value instead of removing them.
